pulp_smash/tests/rpm/api_v2/test_repo_create.py

```diff
@@ -0,0 +1,81 @@
+# coding=utf-8
+"""Test the API endpoints for RPM `repositories`_.
+
+The assumptions explored in this module have the following dependencies::
+
+    # test_create_sync.py
+    It is possible to create an RPM repo.
+    ├── It is impossible to create a duplicate relative url.
+    ├── It is possible to create an RPM repo with feed url. ← TEST THIS
+        ├── It is possible to sync repository from feed. ← TEST THIS
+    ├── It is possible to create an RPM repo without feed url.
+        ├── It is possible to upload single RPM file.
+        ├── It is possible to upload directory of RPM files.
+        ├── It is possible to upload ISO of RPM files.
+
+    # test_distribute.py
+    It is posible to publish an RPM repo with the "???" distributor.
+    ...
+
+.. _repositories:
+   http://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/repo/cud.html
+
+"""
+from __future__ import unicode_literals
+
+import requests
+from pulp_smash.config import get_config
+from pulp_smash.constants import REPOSITORY_PATH
+from random import randint
+from unittest2 import TestCase
+
+
+_RPM_REPO_URL = 'http://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/zoo/'
+
+
+def _rand_str():
+    """Return a randomized string."""
+    return type('')(randint(-999999, 999999))
+
+
+class CreateFeedTestCase(TestCase):
```

Comment:
We definitely want to do similar tests for both yum_importer and puppet_importer. 

In both cases we want to have tests that are using a feed. The sync task needs to be monitored via API and assertions need to be made about the task status and error field. These tests need to assert that a publish task was also spawned after the sync finishes. The publish task should also be monitored and the same assertions should be made about its status and errors. Then a few pieces of the published content should be retrieved via HTTP request from the published repository to make sure that it is actually published. 

We also need tests that check how uploading functionality works for RPMs and Puppet modules. This workflow is documented here [0]. All upload tests should include an API call to publish the newly uploaded content. Then the content should be retrieved via HTTP request to assert that it got published. 

[0] http://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/content/upload.html